antarest/study/storage/variantstudy/model/command/generate_thermal_cluster_timeseries.py

```diff
@@ -0,0 +1,149 @@
+                )["data"]
+                modulation_capacity = np.array([row[2] for row in modulation_matrix])
+                ts_generator_matrix = study_data.tree.get(
+                    ["input", "thermal", "prepro", area_id, thermal.id.lower(), "data"]
+                )["data"]
+                fo_duration, po_duration, fo_rate, po_rate, npo_min, npo_max = [
+                    np.array(col, dtype=int) if i not in [2, 3] else np.array(col, dtype=float)
+                    for i, col in enumerate(zip(*ts_generator_matrix))
+                ]
+                cluster = ThermalCluster(
+                    unit_count=thermal.unit_count,
+                    nominal_power=thermal.nominal_capacity,
+                    modulation=modulation_capacity,
+                    fo_law=ProbabilityLaw(thermal.law_forced.value.upper()),
+                    fo_volatility=thermal.volatility_forced,
+                    po_law=ProbabilityLaw(thermal.law_planned.value.upper()),
+                    po_volatility=thermal.volatility_planned,
+                    fo_duration=fo_duration,
+                    fo_rate=fo_rate,
+                    po_duration=po_duration,
+                    po_rate=po_rate,
+                    npo_min=npo_min,
+                    npo_max=npo_max,
+                )
+                # 7- Generate the time-series
+                results = generator.generate_time_series(cluster, nb_years)
+                generated_matrix = results.available_power.T.tolist()
+                # 8- Generates the UUID for the `get_inner_matrices` method
+                uuid = study_data.tree.context.matrix.create(generated_matrix)
+                self._INNER_MATRICES.append(uuid)
+                # 9- Write the matrix inside the input folder.
+                df = pd.DataFrame(data=generated_matrix, dtype=int)
+                target_path = self._build_matrix_path(tmp_path / area_id / thermal.id.lower())
+                df.to_csv(target_path, sep="\t", header=False, index=False)
+
+    def to_dto(self) -> CommandDTO:
+        return CommandDTO(action=self.command_name.value, args={})
+
+    def match_signature(self) -> str:
+        return str(self.command_name.value)
+
+    def match(self, other: "ICommand", equal: bool = False) -> bool:
+        # Only used inside the cli app that no one uses I believe.
+        if not isinstance(other, GenerateThermalClusterTimeSeries):
+            return False
+        return True
+
+    def _create_diff(self, other: "ICommand") -> t.List["ICommand"]:
+        # Only used inside the cli app that no one uses I believe.
+        raise NotImplementedError()
+
+    def get_inner_matrices(self) -> t.List[str]:
+        # This is used to get used matrices and not remove them inside the garbage collector loop.
+        return self._INNER_MATRICES
+
+    @staticmethod
+    def _replace_safely_original_files(study_path: Path, tmp_path: Path) -> None:
+        backup_dir = Path(
```

Comment:
Not clear to me what the backup dir is for, it seems that we never restore it:
should we add a try / catch to try to restore it if we get an exception when we perform the move operation ?

Reply:
Indeed we could do that